### DISCUSSION TITLE
bold and bold-italic don't show correctly for "JetBrains Mono" on linux

### DISCUSSION THREAD

**Discussion by smjt2000** [ghostty-org/ghostty#3935] [Feature Requests, Ideas]
Hi.
English is not my native language.
I've using Konsole for a long time and configured it as I want and it's look very nice to me. I heard about ghostty and just want to try it and start to use it as my primary terminal emulator.
I use starship on bash and see that some colors and fonts are not displayed correctly.
here is the comparison for konsole and ghostty:
Left window is konsole and right is ghostty.

![Screenshot_20241229_232207](https://github.com/user-attachments/assets/1d94289f-82d5-4f06-90ef-a3ed190b13cc)

You can see that color of prompt is not the same and on konsole it's bold but on ghostty it's not.
Also in Vim, text is bold-italic and in  powerline "Normal" and "SPELL [EN_US]" is bold-italic on konsole.

I set the theme for konsole, ghostty and vim to ChallengerDeep. but I see that colors are not exactly the same on this terminal emulators.

my config file is this:
```
font-size = 12

font-family = JetBrains Mono
font-family-bold = JetBrains Mono Bold
font-family-italic= JetBrains Mono Italic
font-family-bold-italic = JetBrains Mono Bold Italic

font-style-bold = Bold
font-style-italic = Italic
font-style-bold-italic = Bold Italic

font-variation-bold = wght=700
font-variation-italic = wght=700
font-variation-bold-italic = wght=700

bold-is-bright = true

theme = ChallengerDeep

cursor-style = bar

mouse-hide-while-typing = true

copy-on-select = false
```

if I set `bold-is-bright = false` the prompt color will be blueish, but it should be purplish.

ghostty information(pacman):
```
$ ghostty --version
Ghostty 1.0.0-dev+0000000

Version
  - version: 1.0.0-dev+0000000
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.7
    runtime    : 4.16.7
  - libadwaita : enabled
    build      : 1.6.2
    runtime    : 1.6.2
```

**Comment by polirritmico** [comment]
Maybe we are having the same issue. Also KDE here: https://github.com/ghostty-org/ghostty/discussions/3588

**Reply by smjt2000** [reply]
Yeah we have almost the same issue.
I didn't try any other font except "Vazirmatn"(font for Persian and Arabic) but it seems ghossty ignores this font and use an ugly font if I do this:
```
font-family = ""
font-family = "Vazirmatn"
...
```

**Comment by 00-kat** [comment]
Try remove these three lines:
```ini
font-family-bold = JetBrains Mono Bold
font-family-italic= JetBrains Mono Italic
font-family-bold-italic = JetBrains Mono Bold Italic
```

Also worth noting that JetBrains Mono is the default font Ghostty uses, so you don't need to specify `font-family` if you don't want to.

**Reply by smjt2000** [reply]
I remove them and now the bold font is not that bold anymore! it's like bold when I set these options and semi-bold without them.
see "NORMAL" in powerline.
![Screenshot_20241230_124556](https://github.com/user-attachments/assets/5378cb09-7ae8-4061-8968-97893dd0dd4f)
![Screenshot_20241230_124634](https://github.com/user-attachments/assets/28b88978-7d39-4635-bfe3-76b628926d85)

**Reply by doucol** [reply]
> Try remove these three lines:
> 
> ```ini
> font-family-bold = JetBrains Mono Bold
> font-family-italic= JetBrains Mono Italic
> font-family-bold-italic = JetBrains Mono Bold Italic
> ```
> 
> Also worth noting that JetBrains Mono is the default font Ghostty uses, so you don't need to specify `font-family` if you don't want to.

I'm having this problem as well - and this does indeed fix it. But, I'm curious if this is a bug or if I've done something wrong with the setup / config?

**Comment by smjt2000** [comment]
I check my config with Qtile and hyprland just to make sure it's not because plasma and there is no difference.

**Reply by agiUnderground** [reply]
on macos I also can't set font to bold.

**Reply by shayne008** [reply]
I am having the same issue on macOS

**Reply by rshemtov13** [reply]
me too

**Reply by anuragkumar2921** [reply]
me too

**Reply by 00-kat** [reply]
If you're also having this issue, please consider using the upvote button, or alternatively the :+1: reaction.  Every time someone sends "me too", everyone subscribed to the thread gets a new email, fyi.

**Comment by raviverman** [comment]
Can be due to `TERM` env var. Can you try this?
`TERM=xterm-256color nvim .config/starship.toml`

**Reply by smjt2000** [reply]
I've set this in my bashrc.
`export TERM=xterm-256color`

**Comment by ghost** [comment]
Hello. I am on NixOS 25.05 + hyprland.
tested with: 2 nerd fonts: Lilex & Jetbrains-mono
In `kitty.conf` i simply set
`font_family JetBrainsMono NF ExtraLight` and every font variation looks as expected.

In ghostty, when i specify `font-family = "JetBrainsMono NF Light"`
italics works, but bold font has style of (plain) JetBrainsMono NF Light.
If i also explicitly specify `font-family-bold = "JetBrainsMono NF SemiBold"`
normal, bold and italic text looks ok. (issue with bold style gets fixed).
But there is also an issue with bold-italic text which has style of (plain) italics.
setting `font-family-bold-italic = "JetBrainsMono NF SemiBold Italic"`
or anything similar to this (like ExtraBold or Bold) does not help. Both Lilex and Jetbrains fonts are not getting bold, just italic, where they must be bold-italic.

tmux and $TERM are not involved in the problem.

I have also had similar font issues on Arch a few weeks ago before jumping to NixOS.

Here is my ghostty config:
```
font-size = 13.5
# font-family = "Lilex Nerd Font"
# font-family-bold = "Lilex Nerd Font Bold"
# font-family-bold-italic = "Lilex Nerd Font Bold Italic"
font-family = "JetBrainsMono NF Light"
font-family-bold = "JetBrainsMono NF SemiBold"
font-family-bold-italic = "JetBrainsMono NF SemiBold Italic"
cursor-style-blink = false
confirm-close-surface = false
```
kitty (first image):
![image](https://github.com/user-attachments/assets/e27f754f-8618-458e-a211-934f9198aa6e)
ghostty (second image):
![image](https://github.com/user-attachments/assets/86040ef7-732f-4ff9-8ea0-184e0eccd3b7)

While i was testing and writing all this i've also noticed that `font-family-bold = "JetBrainsMono NF Bold"` does not make text bold. Only SemiBold and ExtraBold do.

```
ghostty --version
Ghostty 1.1.3
Version
  - version: 1.1.3
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - desktop env: other
  - GTK version:
    build      : 4.18.5
    runtime    : 4.18.5
  - libadwaita : enabled
    build      : 1.7.2
    runtime    : 1.7.2
  - libX11     : enabled
  - libwayland : enabled
```


Output of `ghostty +list-fonts | grep Jet`:
```
JetBrainsMono Nerd Font
JetBrainsMono NF Bold
JetBrainsMono NF Bold Italic
JetBrainsMono NF ExtraBold
JetBrainsMono NF ExtraBold Italic
JetBrainsMono NF ExtraLight
JetBrainsMono NF ExtraLight Italic
JetBrainsMono NF Italic
JetBrainsMono NF Light
JetBrainsMono NF Light Italic
JetBrainsMono NF Medium
JetBrainsMono NF Medium Italic
JetBrainsMono NF Regular
JetBrainsMono NF SemiBold
JetBrainsMono NF SemiBold Italic
JetBrainsMono NF Thin
JetBrainsMono NF Thin Italic
```

Edit: added images.
Edit 2: clarification of screenshots:
`# Compile the project in the current directory:` (italic) (this and the next line are part of `curl cht.sh/zig` output)
`zig build` (normal)
`~/path` (bold-italic)
`>` (bold) `ls` (normal) 
`config` (normal) `example_dir` (bold)
